massiv/tests/Data/Massiv/Array/StencilSpec.hs

```diff
@@ -102,3 +102,51 @@ spec = do
         stencilCorners (2 :. 2) (-2 :. -2) arr `shouldBe` [[0, 0, 0], [0, 0, 0], [0, 0, 1]]
       it "Direction Left/Bottom Corner" $
         stencilCorners (2 :. 0) (-2 :. 2) arr `shouldBe` [[0, 0, 0], [0, 0, 0], [3, 0, 0]]
+    describe "reformDW" $ do
+      it "map stencil with stride on small array" $
+        let kernel =
+                [[-1, 0, 1], [0, 1, 0], [-1, 0, 1]] :: Array U Ix2 Int
+            stencil = makeConvolutionStencilFromKernel kernel
+            stride = 2
+            strideArr =
+                mapStride stride (1 :. 1) $ mapStencil (Fill 0) stencil arr
+        in computeAs U strideArr `shouldBe` [[-4]]
+      it "map stencil with stride on larger array" $
+        let kernel =
+                [[-1, 0, 1], [0, 1, 0], [-1, 0, 1]] :: Array U Ix2 Int
+            stencil = makeConvolutionStencilFromKernel kernel
+            stride = 2
+            largeArr = fromLists' Par
+                [ [5*n + 1 .. 5 * (n + 1)] | n <- [0..4]]
+                  :: Array U Ix2 Int
+            stencilledArr = mapStencil (Fill 0) stencil largeArr
+            strideArr = mapStride stride (2 :. 2) stencilledArr
+         in do
+            print largeArr
+            print $ computeAs U stencilledArr
+            print $ computeAs U strideArr
+            computeAs U strideArr `shouldBe` [[-6, 1], [-13, 9]]
```

Comment:
@CSVdB I am doing some work on mapping with stride and I realized that this test looks wrong to me. Here is application of stencil without `(2 :. 2)` stride:
```
(Array U Seq (5 :. 5)
  [ [ -6,0,1,2,14 ]
  , [ -8,3,4,5,28 ]
  , [ -13,8,9,10,43 ]
  , [ -18,13,14,15,58 ]
  , [ 4,20,21,22,44 ]
  ])
```
Now according to my understanding and the email you sent me about the stride, we want to keep every 2nd row and column when applying the stencil with stride, thus the expected result should be something like that instead:
```
(Array U Seq (3 :. 3)
  [ [ -6,1,14 ]
  , [ -13,9,43 ]
  , [ 4,21,44 ]
  ])
```

Please, correct me if I am wrong. Same argument applies to the test case above this one ("on small array").